src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -51,6 +57,7 @@ public bool RemoveBuffer(long pageNumber)
             //This assumes that we don't replace buffers just set them.
             set
             {
+                value.AddRef();
```

Comment:
Incrementing a reference inside the setter is very tricky. I'm concerned that this can be a potentially error prone construction.

Why not to increment it explicitly in the code where this setter is called?

Reply:
done